File: atcoder/abc145/abc145_c/abc145_c.cpp
#include <bits/stdc++.h>

using namespace std;

int main(void)
{
    int n;
    cin >> n;

    int x[n] = {}, y[n] = {};
    vector<int> v;
    for(int i = 0; i < n; i++)
    {
        v.push_back(i);
        cin >> x[i] >> y[i];
    }


    double ans = 0.0, cnt = 0.0;

    do
    {
        for(int i = 1; i < n; i++)
        {
            //cout << sqrt(1.0*x[v[i]]*x[v[i]] + 1.0*y[v[i]]*y[v[i]]);
            ans += sqrt((x[v[i - 1]] - x[v[i]])*(x[v[i - 1]] - x[v[i]]) + (y[v[i - 1]] - y[v[i]])*(y[v[i - 1]] - y[v[i]]));
        }
        cnt++;
    }
    while (next_permutation(v.begin(), v.end()));
    ans /= cnt;
    cout << ans << endl;
}

File: atcoder/abc145/abc145_c/abc145_c.py
from itertools import permutations
import math

n = int(input())

x = []
y = []
v = []

for i in range(n):
    v.append(i)
    x1, y1 = map(int, input().split())
    x.append(x1)
    y.append(y1)

ans = 0
cnt = 0

for i in list(permutations(v)):
    for f in range(1, n):
        ans += math.sqrt((x[i[f - 1]] - x[i[f]])*(x[i[f - 1]] - x[i[f]]) + (y[i[f - 1]] - y[i[f]])*(y[i[f - 1]] - y[i[f]]))
    cnt += 1

ans /= cnt
print(ans)
